src/libpmem2/map_posix.c

```diff
@@ -319,11 +341,32 @@ pmem2_map(const struct pmem2_config *cfg, const struct pmem2_source *src,
 	const size_t alignment = get_map_alignment(content_length,
 			src_alignment);
 
+	ret = pmem2_config_validate_addr_alignment(src, cfg);
+	if (ret)
+		return ret;
+
+	/* let's get addr and addr_flag from cfg */
+	void *mmap_addr = cfg->addr;
+	int mmap_addr_flag = cfg->addr_flag;
+
+	/* "translate" pmem2 addr flag into linux one */
+	if (cfg->addr_flag == PMEM2_ADDRESS_FIXED_NOREPLACE)
+	/* check if kernel supports MAP_FIXED_NOREPLACE flag */
+#ifdef MAP_FIXED_NOREPLACE
```

Comment:
Should we just silently ignore this at compile time if the machine used to compile doesn't support it? It doesn't seem right.

I'd rather redefine MAP_FIXED_NOREPLACE and fail at runtime if the kernel doesn't support it.

Anyway, add braces.